logstash-core-event-java/src/main/java/com/logstash/Accessors.java

```diff
@@ -67,7 +67,7 @@ private Object findTarget(FieldReference field) {
         target = this.data;
         for (String key : field.getPath()) {
             target = fetch(target, key);
-            if (target == null) {
+            if (! targetIsCollection(target)) {
```

Comment:
this method only checks that the argument is a collection, I suggest renaming it as `isCollection`